fdbserver/ClusterController.actor.cpp

```diff
@@ -4631,3 +4713,66 @@ ACTOR Future<Void> clusterController(Reference<ClusterConnectionFile> connFile,
 		hasConnected = true;
 	}
 }
+
+namespace {
+
+// Tests `ClusterControllerData::updateWorkerHealth()` can update `ClusterControllerData::workerHealth` based on
+// `UpdateWorkerHealth` request correctly.
+TEST_CASE("/fdbserver/clustercontroller/updateWorkerHealth") {
+    // Create a testing ClusterControllerData. Most of the internal states do not matter in this test.
+    state ClusterControllerData data(ClusterControllerFullInterface(),
+                                     LocalityData(),
+                                     ServerCoordinators(Reference<ClusterConnectionFile>(new ClusterConnectionFile())));
+    state NetworkAddress workerAddress(IPAddress(0x01010101), 1);
+    state NetworkAddress badPeer1(IPAddress(0x02020202), 1);
+    state NetworkAddress badPeer2(IPAddress(0x03030303), 1);
+    state NetworkAddress badPeer3(IPAddress(0x04040404), 1);
+
+    // Create a `UpdateWorkerHealthRequest` with two bad peers, and they should appear in the `workerAddress`'s
+    // degradedPeers.
+    {
+        UpdateWorkerHealthRequest req;
+        req.address = workerAddress;
+        req.degradedPeers.push_back(badPeer1);
+        req.degradedPeers.push_back(badPeer2);
+        data.updateWorkerHealth(req);
+        ASSERT(data.workerHealth.find(workerAddress) != data.workerHealth.end());
+        auto& health = data.workerHealth[workerAddress];
+        ASSERT_EQ(health.degradedPeers.size(), 2);
+        ASSERT(health.degradedPeers.find(badPeer1) != health.degradedPeers.end());
+        ASSERT_EQ(health.degradedPeers[badPeer1].startTime, health.degradedPeers[badPeer1].lastRefreshTime);
+        ASSERT(health.degradedPeers.find(badPeer2) != health.degradedPeers.end());
+    }
+
+    // Create a `UpdateWorkerHealthRequest` with two bad peers, one from the previous test and a new one.
+    // The one from the previous test should have lastRefreshTime updated.
+    // The other one from the previous test not included in this test should be removed.
+    {
+        wait(delay(0.001));
```

Comment:
Why do we need this wait here? Also, if we can remove this wait, I guess we don't need the `state` in line 4723-4729?

Reply:
In simulation, without delay, now() returns the same value no matter when you call it. Here, I try to make the time moves a little bit to test `lastRefreshTime` is updated correctly.

I added a comment why we need this.